source/tomopy/misc/corr.py

```diff
@@ -267,6 +268,137 @@ def median_filter_cuda(arr, size=3, axis=0):
     return out
 
 
+def find_nonfinite_values_prj(projection, slc_idx):
+    """Allows the user to easily obtain the Z, x, y coordinates of nonfinite 
+       values for a given 2D projection in a 3D array.
+
+    Parameters
+    ----------
+    projection : 2D nd.array
+        The 2D projection for a given 3D data array.
+
+    slc_idx : int
+        The projection index inside the main 3D array.
+
+    Returns
+    -------
+    The indices for the current projection used, the x coordinates, and y
+    coordinates of all non-finite values.
+    """
+
+    # Determining where the nonfinite values are
+    boolean_of_nonfinite = ~np.isfinite(projection)
+
+    # Obtaining there x, y locations
+    idx_of_nonfinite_prj = np.nonzero(boolean_of_nonfinite)
+
+    # Creating an identical shaped array for the z position in the 3D array the prj comes from
+    idx_of_nonfinite_data = np.asarray(
+        [slc_idx] * len(idx_of_nonfinite_prj[0]))
+
+    return idx_of_nonfinite_data, idx_of_nonfinite_prj[0], idx_of_nonfinite_prj[1]
+
+
+def determine_nonfinite_kernel_idxs(x_idx, y_idx, kernel, shape_x, shape_y):
```

Comment:
```suggestion
def _determine_nonfinite_kernel_idxs(x_idx, y_idx, kernel, shape_x, shape_y):
```
Functions that are not intended for external use should have a leading underscore.